### ISSUE TITLE
Source Filter does not work with directory names

### ISSUE THREAD

**Issue by krallus** [spotbugs/spotbugs#1792]
This issue effectively re-opens #694 as the functionality implemented there is not currently working in SpotBugs.

## Environment Setup
_OS_: Linux Mint 20.1 Cinnamon
_Eclipse_: Version: 2021-09 (4.21.0)
_Spotbugs Plugin_: 4.4.2.r202110112327-c0382dc

_Spotbugs Exclude Filter File_: `/fullpath/spotbugsExcludes.xml`

Contents of `spotbugsExcludes.xml`:
```xml
<?xml version="1.0" encoding="UTF-8"?>
<FindBugsFilter xmlns="https://github.com/spotbugs/filter/3.0.0"
	xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
	xsi:schemaLocation="https://github.com/spotbugs/filter/3.0.0 https://raw.githubusercontent.com/spotbugs/spotbugs/3.1.0/spotbugs/etc/findbugsfilter.xsd">
	<Match>
		<Source name="~.*generated-sources.*"/>
	</Match>
</FindBugsFilter>
```
## Expected Behaviour
Spotbugs will **not** issue warnings on any file located under any directory that includes "generated-sources" anywhere in the path, such as `/pathtoproject/target/generated-sources/wsimport/com/package/subpackage/ArticleInProgress.java`.

## Actual Behaviour
Spotbugs continues to issue warnings for source files located in any path below `/pathtoproject/target/generated-sources/`

## Other Experiments
- These alternatives still **do** (unfortunately) continue to produce warnings on `ArticleInProgress.java`:
  - `name="~.*generated-sources/wsimport/com/package/subpackage/ArticleInProgress.java"`
  - `name="~.*generated-sources.*ArticleInProgress.java"`
  - `name="/pathtoproject/target/generated-sources/wsimport/com/package/subpackage/ArticleInProgress.java"`
  - `name="target/generated-sources/wsimport/com/package/subpackage/ArticleInProgress.java"`
- These alternatives do **not** continue to produce warnings on `ArticleInProgress.java`:
  - `name="~.*ArticleInProgress.java"`
  - `name="~.*/ArticleInProgress.java"`
  - `name="~.*com/package/subpackage/ArticleInProgress.java"`
  - `name="~.*com/package/subpackage/.*"`
    - Note that in this last case, all files are excluded by Spotbugs that appear in the `com.package.subpackage` Java package, even those that are under a source folder other than `/pathtoproject/target/generated-sources/wsimport/`.

## Hypothesis
The Source Filter is only examining a file's source path starting at the root of the particular source folder that it is in, whereas one would expect it to include the path location of the source root as well (e.g. `/pathtoproject/target/generated-sources/wsimport/'), at least starting at the project root, if not the filesystem root.

**Comment by MarcoMartins86** [comment]
I also wanted to use this functionality to filter out generated sources dir with no success. Downloaded the reproducer from https://github.com/spotbugs/spotbugs/issues/694 and changed the pom to use the last spotbugs version available and confirm that this issue is still happening.

**Comment by prdoyle** [comment]
Has anyone found a workaround for this? I'd like to do the same thing.

**Comment by stephanroch** [comment]
I'm not sure if this isn't working as expected. In the reproducer from #694 one has to add in the configuration of the spotbugs maven plugin one line:
`<addSourceDirs>true</addSourceDirs>`
This parameter was introduced together with #694 for the maven plugin in https://github.com/spotbugs/spotbugs-maven-plugin/pull/111/files. Unfortunately there is no integration test for the maven plugin. Core spotbugs has a unit test for the filtering which is working. I managed to use the project of the reproducer in Eclipse IDE with the spotbugs eclipse plugin and if I configure there to use the given exclude file then the generated sources are excluded from the analaysis as expected.

One thing to note - we had some problems defining the exclude pattern: in the end `~.*target.generated-sources.*` worked well, but we tried first with the separator / between target and generated-sources and this does surpisingly not work - maybe because we use Windows, were \\ works as separator which is rather unusual when you think about Maven and Java. Thats why we use one period between the directories of the filter pattern.

Please check your project configuration. Otherwise I would suggest to close this issue.

**Comment by krallus** [comment]
It's been four years since I submitted this issue and so I cannot say for certain that I am recreating my initial conditions, but I just tested with the following:

_OS_: Linux Mint 21.3 Cinnamon
_Eclipse_: Version: 2025-09 (4.37.0)
_JVM_: openjdk version "25.0.1" 2025-10-21
_Spotbugs Plugin_: 4.9.8.r202510181643-c1fa7f2

In this environment, the exact `spotbugsExcludes.xml` I gave in the original description works as expected. I even tested this on the same project from four years ago in which I first noticed this problem. So I'm closing this issue.
